receiver/podmanreceiver/receiver.go

```diff
@@ -87,3 +158,52 @@ func (r *receiver) scrape(context.Context) (pdata.Metrics, error) {
 	}
 	return md, nil
 }
+
+func (r *receiver) handleEvents(ctx context.Context, eventBackoff *backoff.ExponentialBackOff) error {
+	c, err := r.clientFactory(r.set.Logger, r.config)
+	if err != nil {
+		r.set.Logger.Error("error fetching/processing events", zap.Error(err))
+		return err
+	}
+	r.client = c
+
+	// Fetch the response from the endpoint
+	response, fetchErr := r.client.getEventsResponse()
+	if fetchErr != nil {
+		r.set.Logger.Error("Error while fetching events", zap.Error(fetchErr))
+		return fetchErr
+	}
+
+	dec := json.NewDecoder(response.Body)
+	for {
+		// Translate the response to the events format.
+		decodedEvent, decodeErr := decodeEvents(dec)
+		if decodeErr != nil {
+			r.set.Logger.Error("Error decoding the event", zap.Error(decodeErr))
+			if connCloseErr := response.Body.Close(); connCloseErr != nil {
+				r.set.Logger.Error("Error while closing the connection", zap.Error(connCloseErr))
+				return connCloseErr
+			}
+			return decodeErr
+		}
+
+		// Translate the events into the pdata.logs format
+		ld, translationErr := traslateEventsToLogs(r.set.Logger, decodedEvent)
+		if translationErr != nil {
+			r.set.Logger.Error("Error translating event to log", zap.Error(translationErr))
+			return translationErr
+		}
+
+		transferErr := r.logsConsumer.ConsumeLogs(ctx, ld)
+		if transferErr != nil {
+			r.set.Logger.Error("Error transferring it to the next component", zap.Error(transferErr))
+			return transferErr
+		}
+		eventBackoff.Reset()
+		r.shutDownSync.Lock()
+		if r.isLogsShutdown {
+			return nil
+		}
+		r.shutDownSync.Unlock()
```

Comment:
This is dangerous: if `isLogsShutdown` happens, this will never be executed, keeping the mutex locked indefinitely. When the unlock is the last thing to happen in your function, prefer to keep it as `defer r.shutDownSync.Unlock()`, close to where you locked.

If you are only needing the lock to read a value, you can use a `RWMutex` and use a `RLock` here. This way, if you have two concurrent `handleEvents`, one will not block the other if they are both trying to read the value, but they will both block/wait in case something else needs to write to this value.

Reply:
Oh yes, Good catch. Added `r.shutdownSync.unlock()` in the if condition, as we can not use defer because defer will not be called in the successful scenario as the function only returns an error if any.

Reply:
Right, I missed the `for` above. Forget my previous comments about using defer then :-)